### PR TITLE
Refactor to add QCMetric class and list of URLs for reference images/figures

### PR DESCRIPTION
**Pull request by dbirman**

### CODE REVIEW

File: examples/quality_control.json [hunk 1]
```diff
@@ -12,11 +12,18 @@
...
-         "qc_metrics": {
-            "Video_1_num_frames": 662,
-            "Video_2_num_frames": 662,
-            "Frame_match": true
-         },
+         "qc_metrics": [
+            {
+               "name": "Video_frame_match",
+               "value": {
+                  "Video_1_num_frames": 662,
```
Comment (saskiad) [inline examples/quality_control.json:19]: I guess I thought that each metrics would be a separate instance of `qc_metrics`, is that not the idea?

File: src/aind_data_schema/core/quality_control.py [hunk 2]
```diff
@@ -19,22 +19,30 @@ class Status(str, Enum):
     PASS = "Pass"
 
 
+class QCMetric(BaseModel):
+    """Description of a single quality control metric"""
+    name: str = Field(..., title="Metric name")
+    value: Any = Field(..., title="Metric value")
+    description: Optional[str] = Field(None, title="Metric description")
```
Comment (saskiad) [inline src/aind_data_schema/core/quality_control.py:26]: add the `default=` for the optional fields.

**Review by saskiad — CHANGES_REQUESTED** [review]
fix the defaults. Also a quick question on how you envision using it

File: examples/quality_control.json (hunk 1 above)
Comment (dbirman) [reply examples/quality_control.json:19]: Oh I see what you were imagining, so the evaluation here would have two metrics (video 1 frames / video 2 frames) and then the evaluation is passed/failed based on whether they match? (let me know if that's wrong). I'll change the example.

Comment (dbirman) [reply examples/quality_control.json:19]: For evaluations that have rules like this, do we need a place to put that? I feel like "Notes" doesn't imply that there should be a description of the evaluation rule.

**Comment by dbirman** [comment]
Added quality_control.Stage per discussion, and modified examples to better reflect our concept. Also added name to QCEvaluation and made descriptions optional, to separate stage/name/desc from each other.

File: src/aind_data_schema/core/quality_control.py [hunk 3]
```diff
@@ -19,25 +19,46 @@ class Status(str, Enum):
...
 class QCEvaluation(AindModel):
-    """Description of one evaluation stage"""
+    """Description of one evaluation stage, with one or more metrics"""
 
     evaluation_modality: Modality.ONE_OF = Field(..., title="Modality")
-    evaluation_stage: str = Field(..., title="Evaluation stage")
-    evaluator_full_name: str = Field(..., title="Evaluator full name")
+    evaluation_stage: Stage = Field(..., title="Evaluation stage")
+    evaluation_name: str = Field(..., title="Evaluation name")
+    evaluation_desc: Optional[str] = Field(default=None, title="Evaluation description")
```
Comment (saskiad) [inline src/aind_data_schema/core/quality_control.py:47]: can we make this `evaluation_description` - I'm reluctant to shorten things, even obvious things.

File: src/aind_data_schema/core/quality_control.py [hunk 4]
```diff
@@ -19,25 +19,46 @@ class Status(str, Enum):
...
+    evaluator: str = Field(..., title="Evaluator full name")
     evaluation_date: date = Field(..., title="Evaluation date")
-    qc_metrics: AindGenericType = Field(AindGeneric(), title="QC metrics")
+    qc_metrics: List[QCMetric] = Field(title="QC metrics")
```
Comment (saskiad) [inline src/aind_data_schema/core/quality_control.py:50]: I think this needs the `...` in the Field

**Review by saskiad — CHANGES_REQUESTED** [review]
super minor things, but otherwise I think it's good.

**Review by saskiad — APPROVED** [review]

**Pull request merged**